### PR TITLE
Feature/go router routes

### PR DESCRIPTION
**Pull request by schalky**
Integration of go router
Updated handling of links

### CODE REVIEW

File: lib/app/circle/circle_page.dart [hunk 1]
```diff
@@ -187,12 +187,8 @@ class CirclePageState extends ConsumerState<CirclePage> {
     final repo = ref.read(repositoryProvider);
     await repo.activateSession(session: session);
     if (!mounted) return;
-    await Navigator.pushReplacement(
-      context,
-      FadeRoute(
-        page: CircleSessionPage(session: session),
-      ),
-    );
+    debugPrint(session.id);
```
Comment (blopker) [inline lib/app/circle/circle_page.dart:190]: Debug

File: lib/app/login/login_page.dart [hunk 2]
```diff
@@ -50,7 +55,7 @@ class _LoginPanel extends StatelessWidget {
                         label: t.login,
                         width: 294,
                         onPressed: () {
-                          Navigator.pushNamed(context, '/login/phone');
+                          context.go('/login/phone');
```
Comment (blopker) [inline lib/app/login/login_page.dart:58]: Not using the static values from app_routes?

File: lib/app/login/login_page.dart [hunk 3]
```diff
@@ -103,12 +108,13 @@ class _LoginPanel extends StatelessWidget {
   }
 }
 
-class LoginPage extends StatelessWidget {
-  const LoginPage({Key? key}) : super(key: key);
+class WelcomePage extends ConsumerWidget {
```
Comment (blopker) [inline lib/app/login/login_page.dart:111]: Thanks for the rename!

File: lib/app_routes.dart [hunk 4]
```diff
@@ -1,88 +1,149 @@
...
+  String? _pendingRoute;
+
+  GoRouter getRouter(WidgetRef ref) {
+    return GoRouter(
+        observers: [
+          SentryNavigatorObserver()
+        ],
+        routes: <GoRoute>[
+          GoRoute(
+            path: home,
```
Comment (blopker) [inline lib/app_routes.dart:42]: One thing we could use later is have the 'name' property filled out. Then we would use `goNamed` and not have to use the actual URL anywhere, even in the static constants.

Comment (schalky) [reply lib/app_routes.dart:42]: true, can switch to that

File: lib/app_routes.dart [hunk 5]
```diff
@@ -1,88 +1,149 @@
...
+                path: 'create',
+                pageBuilder: (context, state) => const MaterialPage(
+                    child: CircleCreateSnapPage(), fullscreenDialog: true),
+              ),
+            ],
+          ),
+          GoRoute(
+            path: dev,
+            pageBuilder: (context, state) =>
+                const MaterialPage(child: DevPage()),
```
Comment (blopker) [inline lib/app_routes.dart:73]: One day I'll understand why some of these widgets need to be wrapped in a Page (DevPage) and some do not (UserProfilePage). Today is not that day.

File: lib/app_routes.dart [hunk 6]
```diff
@@ -1,88 +1,149 @@
...
+          // if the route is a circle, then preserve the route and redirect after
+          // login
+          if (!loggedIn) {
+            _pendingRoute = state.location;
+            return login;
+          }
+          // if this is a snap circle link, redirect to the circle page
+          if (state.subloc == home && state.queryParams.containsKey('snap')) {
+            return '${home}circle/${state.queryParams['snap']}';
+          }
```
Comment (blopker) [inline lib/app_routes.dart:126]: We'll probably want to extract this to another dynamic link handler route at some point.

Comment (schalky) [reply lib/app_routes.dart:126]: probably should just be:             pageBuilder: (context, state) => const NoTransitionPage(child: DevPage()),
i made that change now in my other branch

Comment (schalky) [reply lib/app_routes.dart:126]: we'd have to change the cloud function generator to not use the home path

File: lib/app_routes.dart (hunk 5 above)
Comment (schalky) [reply lib/app_routes.dart:73]: probably should just be: pageBuilder: (context, state) => const NoTransitionPage(child: DevPage()),
i made that change now in my other branch

File: lib/app/login/login_page.dart (hunk 2 above)
Comment (schalky) [reply lib/app/login/login_page.dart:58]: i think i missed that one

**Review by blopker — APPROVED** [review]
Ship it!

**Pull request merged**